src/main/java/org/generationcp/middleware/util/Util.java

```diff
@@ -490,4 +497,27 @@ public static boolean isLeapYear(final int year) {
 		return isLeapYear;
 	}
 
+
+	public static boolean isDateMatchPattern(final String value, final String dateFormat) {
+		final Matcher matcher;
+		if (Util.DATE_AS_NUMBER_FORMAT_KSU.equals(dateFormat)) {// d/M/yy|yyyy
+			matcher = ksuDatePattern.matcher(value);
+		} else {// yyyyyMMdd
+			matcher = defaultDatePattern.matcher(value);
+		}
+		return matcher.matches();
+	}
+
+	public static Date tryParseDateAccurately(final String date, final String format) {
+		try {
+			if (Util.isDateMatchPattern(date, format)) {
```

Comment:
I wonder if we need this check here if date matches pattern. Without it, there will be a parse/null exception which will return null anyway

Reply:
@darla-leafnode, 
I think this can be removed and just use Util.parseDate. This is originally meant to check pattern yyMMdd, some data passes this pattern even if it's pattern is dd/MM/yy and produced wrong output.